docs/pages/build-reference/e2e-tests.mdx

```diff
@@ -97,103 +60,146 @@ appId: dev.expo.eastestsexample # This is an example app id. Replace it with you
+This workflow builds an **.apk** for Android using the `e2e-test` build profile from the previous step. Then it runs the **.maestro/home.yml** flow on the built APK.
 
-<Terminal cmd={['$ eas build --profile build-and-maestro-test']} />
+Here's an example of the same test workflow for iOS:
 
-<ContentSpotlight
-  alt="Logs of successful Maestro tests"
-  src="/static/images/eas-build/tests/03-logs.png"
-  className="max-w-[600px]"
-/>
+```yaml .eas/workflows/e2e-test.yml
+name: e2e-test-ios
 
-When the flow fails, any Maestro artifacts are automatically uploaded as build artifacts. This includes screenshots saved at `~/.maestro/tests` (the default destination). You can download them from the build page.
+on:
+  pull_request:
+    branches: ['*']
 
-<ContentSpotlight
-  alt="Failed Maestro tests artifacts"
-  src="/static/images/eas-build/tests/04-artifacts.png"
-  className="max-w-[600px]"
-/>
-</Step>
+jobs:
+  build_ios_for_e2e:
+    type: build
+    params:
+      platform: ios
+      profile: e2e-test # your eas build profile for E2E test
 
-## More
+  maestro_test:
+    needs: [build_ios_for_e2e]
+    type: maestro
+    params:
+      build_id: ${{ needs.build_ios_for_e2e.outputs.build_id }}
+      flow_path: ['.maestro/home.yml']
+```
 
-If you want to build more advanced custom builds workflows, see the [custom build schema reference](/custom-builds/schema/) for more information.
+Learn more about [Syntx for EAS Workflows](https://docs.eas.dev/workflows/syntax).
 
-To learn more about Maestro flows and how to write them, see the [Maestro documentation](https://maestro.mobile.dev).
+</Step>
 
-## Troubleshooting
+<Step label="6">
+## Run the E2E test workflow
 
-If you encounter the following error message when starting an Android Emulator for E2E tests on EAS Build:
+You can run the E2E test workflow in two ways:
 
-```text
-Failed to configure emulator: emulator with required ID not found.
+1. **Manually using the EAS CLI**
+
+```bash
+npx eas-cli@latest workflow:run .eas/workflows/e2e-test-android.yaml
 ```
 
-This is likely because the New Build Infrastructure is enabled for your project. Go to [**Project Settings**](https://expo.dev/accounts/[account]/projects/[project]/settings) and disable **New Android Builds Infrastructure**.
+2. **Automatically when you open a pull request**
+
+The workflow uses a `pull_request` trigger to run automatically when someone opens a pull request to your repository. Learn more about [EAS Workflow triggers](/eas/workflows/syntax/#on).
+
+After the workflow starts, you can track its progress and view the results in the EAS Dashboard. Here's a screenshot of a completed workflow run:
+
+<ContentSpotlight src="/static/images/eas-build/tests/e2e-workflow.png" />
+
+</Step>
+
+## More
+
+<BoxLink
+  title="Syntax for EAS Workflows"
+  description="Learn more about the syntax for EAS Workflows."
+  href="/eas/workflows/syntax/"
+  Icon={Grid01Icon}
```

Comment:
```suggestion
  Icon={Dataflow01Icon}
```

Let's replace Grid icon with the above since that represents the EAS Workflows icon. You'll have to import it from: `import { Dataflow01Icon } from '@expo/styleguide-icons/outline/Dataflow01Icon';`